### PR TITLE
chore: remove set-output

### PR DESCRIPTION
**Pull request by ryanoolala**
[Notion link](https://www.notion.so/sally-wallet/Refactor-Deprecated-Github-Actions-99d665db11a44bb29261d11ccc1e9c2b?pvs=4) 

Changed set-output for GITHUB_ENV

-   [x] I've kept this PR as small as possible (~600 lines) by splitting it into PRs with manageable chunks of code
-   [x] I've requested reviews from 1-2 reviewers

### CODE REVIEW

File: .github/workflows/build_branch_preview.yml [hunk 1]
```diff
@@ -122,37 +122,35 @@ jobs:
         env:
           NPM_TOKEN: ${{ secrets.GH_TOKEN_SUPPLYALLY_BOT }}
       - name: Get Latest Binary Version # Binary Version will be x.x.x based on the latest tag
-        id: latestBinaryVersion
         run: |
           # Release tag finds the latest tag in the tree branch - i.e. prod-x.x.x
           RELEASE_TAG=$(echo $(git describe --tags --abbrev=0))
           # Using param substitution, we output x.x.x instead
-          echo "::set-output name=version::${RELEASE_TAG#*-}"
+          echo "LATEST_BUILD_VERSION=${RELEASE_TAG#*-}" >> "$GITHUB_ENV"
```
Comment (whoiskai) [inline .github/workflows/build_branch_preview.yml:129]: should this be `LATEST_BINARY_VERSION`?

File: .github/workflows/build_staging.yml [hunk 2]
```diff
@@ -108,34 +108,33 @@ jobs:
         env:
           NPM_TOKEN: ${{ secrets.GH_TOKEN_SUPPLYALLY_BOT }}
       - name: Get Latest Binary Version # Binary Version will be x.x.x based on the latest tag
-        id: latestBinaryVersion
         run: |
           # Release tag finds the latest tag in the tree branch - i.e. prod-x.x.x
           RELEASE_TAG=$(echo $(git describe --tags --abbrev=0))
           # Using param substitution, we output x.x.x instead
-          echo "::set-output name=version::${RELEASE_TAG#*-}"
+          echo "LATEST_BUILD_VERSION=${RELEASE_TAG#*-}" >> "$GITHUB_ENV"
```
Comment (whoiskai) [inline .github/workflows/build_staging.yml:115]: should this be `LATEST_BINARY_VERSION`?

File: .github/workflows/build_branch_preview.yml (hunk 1 above)
Comment (ryanoolala) [reply .github/workflows/build_branch_preview.yml:129]: ah god

File: .github/workflows/build_staging.yml (hunk 2 above)
Comment (ryanoolala) [reply .github/workflows/build_staging.yml]: fix

File: .github/workflows/build_branch_preview.yml (hunk 1 above)
Comment (ryanoolala) [reply .github/workflows/build_branch_preview.yml]: fix

**Review by whoiskai — APPROVED** [review]
LGTM

**Pull request merged**